Add OAuth to application so other teams can add it

thelowlypeon opened the pull request:
This adds a new Grape API that solely handles auth stuff. It stores resulting tokens in a `Team` model, which is now linked to a user

thelowlypeon commented:
sorry for the many updates post PR creation. required some in-the-wild testing because slack won't hit my localhost.

this is ready to go 👍

andygiannini commented on db/connect.rb:
```diff
@@ -3,6 +3,9 @@
     DB = Sequel.connect('sqlite://test/stations_bot-test.db')
+  when :development
+    require 'mysql2'
+    DB = Sequel.connect(ENV['DATABASE_URL'])
```
We should probably clean up these database connections, too. I won't block this PR on it, though.

andygiannini commented on oauth.rb:
```diff
@@ -0,0 +1,42 @@
+          redirect "https://stationtostationapp.com/slack?success=false"
+        end
+      else
+        redirect "https://stationtostationapp.com/slack?success=false"
```
Could we simplify this a bit to something like:

```
      success = false
      if response = RestClient.get('https://slack.com/api/oauth.access', {...})
        json = JSON.parse(response.body)
        if json.key?('access_token') && json.key?('team_id')
          ...
          success = true
        end
      end
      redirect "https://stationtostationapp.com/slack?success=#{success}"
```

thelowlypeon commented:
@andygiannini yikers, this required some more work than i'd expected.

why? because i was an idiot and removed the token from `.env` so as to keep it out of the repo, but i didn't have it on my local machine. i kept getting 401 unauth, but that's because the station to station api was returning that status :(

anyway, addressed your suggestions

thelowlypeon commented:
one small comment -- i keep getting this failure, but it only happens with sqlite. can you give this a pull and try testing?

```
  1) Failure:
SavePlaceCommandTest#test_saves_new_place [/Users/peter/Sites/stations-bot/test/save_place_command_test.rb:13]:
Failed to save place: {"text":"Unknown error SQLite3::SQLException: no such table: main.users_backup0"}.
Expected: 1
  Actual: 0
```

thelowlypeon commented:
@andygiannini fixed that sqlite error i was getting. i forgot that sqlite doesn't support foreign key constraints, and apparently `sequel` doesn't handle it well

andygiannini commented:
Nice fixes for the DB and tests! 👍
